terminal/bundles/org.eclipse.terminal.view.ui/src/org/eclipse/terminal/view/ui/internal/ConsoleManager.java

```diff
@@ -209,18 +195,23 @@ public Optional<ITerminalsView> findConsoleView(TerminalViewId tvid) {
 	 *
 	 * @return The terminals console view instance or <code>null</code> if not found.
 	 */
-	private IViewPart getTerminalsViewWithSecondaryId(TerminalViewId tvid, boolean restore) {
-		for (IViewReference ref : getActiveWorkbenchPage().getViewReferences()) {
+	private Optional<ITerminalsView> findTerminalsViewWithSecondaryId(TerminalViewId tvid, boolean restore) {
+		IWorkbenchPage page = getActiveWorkbenchPage();
+		if (page == null) {
```

Comment:
Is this something we would expect or is it actually indicating an error? e.g. should we` throw noActivePage();`

Reply:
Currently it should not fail here, as we have interface method `public Optional<ITerminalsView> findConsoleView(TerminalViewId tvid)`